autogoal/contrib/ensemble/_stacking.py

```diff
@@ -0,0 +1,100 @@
+# from autogoal.contrib import ensemble
+import numpy as np
+from sklearn.svm import SVC
+from sklearn.naive_bayes import GaussianNB
+from sklearn.linear_model import LogisticRegression
+from sklearn.tree import DecisionTreeClassifier
+
+class StackingEnsemble:
+
+    def __init__(self, 
+    layers = [
```

Comment:
We should also have an option to receive these `layers` and `final` as parameters.

Reply:
What do you mean by that ? 
Because in here in here when creating an instance of the class, the user can add any kind of layer he wants.

Reply:
Yeah, I see now that `layers` and `final` are parameters. I got confused by the weird indentation and the initialization :sweat_smile: 
I think is better if we avoid using mutable objects as default values of a parameter. The default values are actually shared among all instances of the same class, so every new instance of `StackingEnsemble` will have the same instance of `SVC`, for example, that's a very subtle source of bugs.

So my suggestion would be to simply not having any default value here, and force the user to decide which algorithms to use in the ensemble.

Otherwise, if you do want to have an optional parameter, what you can do is `layers=None` in the constructor and then inside the method:

```
if layers is None:
    self.layers = [ ... ] # initialize here
```

Reply:
Oh I understand, do you think we could raise an error if layers is empy ?

Reply:
Yeah, I think that works.